src/ModelContextProtocol.AspNetCore/Authentication/McpAuthenticationHandler.cs

```diff
@@ -43,8 +43,7 @@ public async Task<bool> HandleRequestAsync()
             return false;
         }
 
-        await HandleResourceMetadataRequestAsync();
-        return true;
+        return await HandleResourceMetadataRequestAsync();
```

Comment:
It will be about 5% faster if the await keyword is removed, and the async keyword on the wrapping function is removed. Returning the task directly avoids the overhead of the await state machine for that method.